db/db_test.cc

```diff
@@ -4233,7 +4233,9 @@ TEST_F(DBTest, ConcurrentFlushWAL) {
         threads.emplace_back([&] {
           for (size_t i = cnt; i < 2 * cnt; i++) {
             auto istr = std::to_string(i);
-            WriteBatch batch;
+            WriteBatch batch(0 /* reserved_bytes */, 0 /* max_bytes */,
+                             wopt.protection_bytes_per_key,
```

Comment:
As mentioned in test plan, the technique I used to ensure `WriteOptions::protection_bytes_per_key` is always considered was to (1) change the default `WriteOptions::protection_bytes_per_key` to 8, and (2) run `make check` relying on the assert in `WriteImpl()` to catch any code paths that still had integrity protection disabled. While that caught a lot of meaningful things, it also flagged some irrelevant test code paths like this one. I kept changes to those seemingly irrelevant code paths in this PR just so it's easier to redo such testing again.